core/Segment.php

```diff
@@ -350,4 +350,34 @@ public function __toString()
     {
         return (string) $this->getString();
     }
+
+    /**
+     * Combines this segment with another segment condition, if the segment condition is not already
+     * in the segment.
+     *
+     * The combination is naive in that it does not take order of operations into account.
+     *
+     * @param string $segment
+     * @param string $operator The operator to use. Should be either SegmentExpression::AND_DELIMITER
+     *                         or SegmentExpression::OR_DELIMITER.
+     * @param string $segmentCondition The segment condition to add.
+     * @return string
+     * @throws Exception
+     */
+    public static function combine($segment, $operator, $segmentCondition)
+    {
+        if (empty($segment)) {
+            return $segmentCondition;
+        }
+
+        if (empty($segmentCondition)
+            || strpos($segment, $operator . $segmentCondition) !== false
+            || strpos($segment, $segmentCondition . $operator) !== false
+            || $segment === $segmentCondition
```

Comment:
just wondering... here we may need to also check against `urlencoded` values and `urldecoded`? It is this funny thing with segments that they may be urlencoded or not for some reason. No idea whether this would be a thing here during archiving or if they have all same format. Just wanted to mention it.